### PR TITLE
fava: upgrade to 1.20.1

### PR DESCRIPTION
**Pull request by sutra**
#### Description

Upgrade fava from 1.14 to 1.20.1.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.2 21D49 x86_64
Xcode 13.2.1 13C100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by pmetzger** [comment]
@mrdomino ?

**Comment by mrdomino** [comment]
LGTM

File: python/fava/Portfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ homepage                https://beancount.github.io/fava/
 master_sites            pypi:[string index ${python.rootname} 0]/${python.rootname}
 distname                ${python.rootname}-${version}
 
-python.versions         37 38
+python.versions         37 38 39 310
 python.default_version  38
```
Comment (pmetzger) [inline python/fava/Portfile:27]: Might be a good idea to change the default version if it is possible.

File: python/fava/Portfile [hunk 2]
```diff
@@ -23,7 +23,7 @@ homepage                https://beancount.github.io/fava/
 master_sites            pypi:[string index ${python.rootname} 0]/${python.rootname}
 distname                ${python.rootname}-${version}
```
Comment (reneeotten) [inline python/fava/Portfile:24]: `master_sites` and `distname` can be removed, these are the defaults set by the `python` PG

**Comment by sutra** [comment]
And more ports have to update to support python-310: `py-cheroot`, `py-flask-babel`, `py-markdown2`.

**Comment by mrdomino** [comment]
those might all be easy upgrades, same usual suspects as when this got added for py39 and they only supported up to py38

**Comment by sutra** [comment]
> And more ports have to update to support python310: `py-cheroot`, `py-flask-babel`, `py-markdown2`.

See #13879, #13880, #13881

**Comment by pmetzger** [comment]
Please do squash your commits; the maintainers prefer to merge a single update commit into the history.

**Comment by pmetzger** [comment]
@sutra Please do squash your commits. Thanks.

**Comment by sutra** [comment]
Squashed: #13903

**Comment by pmetzger** [comment]
FYI, generally, you should just force push the squashed version so that it appears under the same merge request (along with the conversation that lead to the final commit...)

**Pull request closed without merging**